src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -100,20 +101,43 @@ namespace {
 NetworkInfo sNetworks[kMaxNetworks];
 } // namespace
 
-// TODO: We cannot get length of operational dataset here, should be fixed after we have byte string type.
-// Note: The codegen for OCTET_STRING will be implemented later since no other clusters are using OCTET_STRING.
 EmberAfNetworkCommissioningError OnAddThreadNetworkCommandCallbackInternal(app::Command *, EndpointId,
-                                                                           const uint8_t * operationalDataset, uint64_t breadcrumb,
+                                                                           const uint8_t * operationalDataset,
+                                                                           size_t operationalDatasetLen, uint64_t breadcrumb,
                                                                            uint32_t timeoutMs)
 {
-    // TODO: Implement AddThreadNetworkCommand after OCTET_STRING is supported.
-    return EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_NETWORK_NOT_FOUND;
+    EmberAfNetworkCommissioningError err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED;
+
+    for (size_t i = 0; i < kMaxNetworks; i++)
+    {
+        if (sNetworks[i].mNetworkType == NetworkType::kUndefined)
+        {
+            VerifyOrExit(operationalDatasetLen <= kMaxThreadDatasetLen, err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
+
+            memcpy(sNetworks[i].mData.mThread.mDataset, operationalDataset, operationalDatasetLen);
+            sNetworks[i].mData.mThread.mDatasetLen = static_cast<uint8_t>(operationalDatasetLen);
+            memcpy(sNetworks[i].mNetworkID, kThreadNetworkID, strlen(kThreadNetworkID));
```

Comment:
What guarantees that buffer is big enough and will remain big enough as the code changes?  Or put another way, should we at least check that the string is not too long to fit in there?

Reply:
Thread network's network id is its extended PAN ID. And the spec defined how it stores in this attribute.
> XPAN ID is a big-endian 64-bit unsigned number, represented on the first 8 octets of the octet string.

Reply:
Um... we are putting way more than 8 octets in there, no?

Reply:
Renamed this constant to `kTemporaryThreadNetworkId`.
The network id of a thread network should be its ext panid, however, we are missing the api for getting its extpanid from dataset tlvs. Added comment about this.